lib/cisco_node_utils/router_ospf_area.rb

```diff
@@ -0,0 +1,290 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ipaddr'
+require_relative 'node_util'
+require_relative 'router_ospf'
+require_relative 'router_ospf_vrf'
+
+module Cisco
+  # node_utils class for ospf_area
+  class RouterOspfArea < NodeUtil
+    attr_reader :router, :vrf, :area_id
+
+    def initialize(ospf_router, vrf_name, area_id, instantiate=true)
+      fail TypeError unless ospf_router.is_a?(String)
+      fail TypeError unless vrf_name.is_a?(String)
+      fail TypeError unless area_id.is_a?(String)
+      fail ArgumentError unless ospf_router.length > 0
+      fail ArgumentError unless vrf_name.length > 0
+      fail ArgumentError unless area_id.length > 0
+
+      # check the area_id is a proper ipv4 address
+      fail ArgumentError if (begin
+                               IPAddr.new(area_id)
+                             rescue
+                               nil
+                             end).nil?
+      fail ArgumentError unless IPAddr.new(area_id).ipv4?
+
+      @router = ospf_router
+      @vrf = vrf_name
+      @area_id = area_id
+
+      set_args_keys_default
+      create if instantiate
+    end
+
+    def self.areas
+      hash_final = {}
+      RouterOspf.routers.each do |instance|
+        name = instance[0]
+        # get all area ids under default vrf
+        area_ids = config_get('ospf_area', 'areas', name: name)
+        unless area_ids.nil?
+          area_ids.uniq.each do |area|
+            hash_final[name]['default'][area] =
+              RouterOspfArea.new(name, vrf, area, false)
+          end
+        end
+        vrf_ids = config_get('ospf', 'vrf', name: name)
+        next if vrf_ids.nil?
+        vrf_ids.each do |vrf|
+          # get all area ids under each vrf
+          area_ids = config_get('ospf_area', 'areas', name: name, vrf: vrf)
+          next if area_ids.nil?
+          area_ids.uniq.each do |area|
+            hash_final[name][vrf][area] =
+              RouterOspfArea.new(name, vrf, area, false)
+          end
+        end
+      end
+      hash_final
+    end
+
+    # Helper method to delete @set_args hash keys
+    def set_args_keys_default
+      @set_args = { name: @router }
+      @set_args[:vrf] = @vrf unless @vrf == 'default'
+      @set_args[:area] = @area_id
+      @get_args = @set_args
+    end
+
+    # rubocop:disable Style/AccessorMethodName
+    def set_args_keys(hash={})
+      set_args_keys_default
+      @set_args = @get_args.merge!(hash) unless hash.empty?
+    end
+
+    def create
+      # create RouterOspfVrf only
+      # area_id is used at each config cli, not as a context
+      # ex: area 1.1.1.1 default-cost 1000
+      # area 1.1.1.1 authentication message-digest
+      RouterOspfVrf.new(@router, @vrf)
+    end
+
+    def destroy
```

Comment:
We can talk about this more in the parking lot today, but I would have expected `destroy` to remove the area, not just reset the area properties back to default.